earthpy/spatial.py

```diff
@@ -573,7 +601,8 @@ def draw_legend(im, classes, titles, bbox=(1.05, 1), loc=2):
     colors = [im.cmap(im.norm(aclass)) for aclass in classes]
 
     patches = [mpatches.Patch(color=colors[i],
-                              label="{l}".format(l=titles[i])) for i in range(len(titles))]
+                              label="{l}".
+                              format(l=titles[i])) for i in range(len(titles))]
```

Comment:
This 3 line statement has a lot going on. Possible to break it up somehow and improve clarity with good names?

Reply:
i don't think we can easily break it up .. maybe i could pull out some of the nested list comprehensions tho!? 

welcome to the backend of matplotlib. but we could def create better var names! This is creating a legend. each patch is a square in the legend. each title is a legend "category" think about say a landuse map with categories like agriculture, urban, etc. each of those words is a title. each box has a color associated with that category, etc. hmmm lemme think about this. this function needs to be rewritten as it actually doesn't work now.

Reply:
Ugh matplotlib... :nauseated_face:

Reply:
Because this is a can of worms - i've added this note to the [issue here](https://github.com/earthlab/earthpy/issues/26) to ensure it is addressed. I don't think i can address it in this PR however. matplotlib is such a beast. i.cant.even.